### ISSUE TITLE
[Bug] Kong failures when Ingress has no Path

### ISSUE THREAD

**Issue by leosunmo** [Kong/kubernetes-ingress-controller#298]
### Summary

If your Ingress resources do _not_ specify an explicit Path under `spec.rules.http.paths` the Kong configuration call from the Kubernetes Ingress Controller fails.
This is something I encountered when installing Jenkins X as their Ingress generating controller (https://github.com/jenkins-x/exposecontroller) does not add `path: /` by default.

**Kong Ingress controller version**
0.4.0

**Kong or Kong Enterprise version**
1.1.2

### Kubernetes version
1.10.11

### Environment
Kops on AWS

### What happened
Getting the following error message spammed for multiple Ingress and non-ingress resources in my cluster:
```
E0528 21:44:53.194218       5 controller.go:117] unexpected failure updating Kong configuration: 
1 errors occured:
    while processing event: {Create} failed: 400 Bad Request {"message":"schema violation (paths.1: length must be at least 1)","name":"schema violation","fields":{"paths":["length must be at least 1"]},"code":2}
```

### Expected behvaior
The Ingress Controller should configure my Ingress in Kong. According to K8s spec the Ingress resource doesn't require an explicit Path, see example: https://kubernetes.io/docs/concepts/services-networking/ingress/#name-based-virtual-hosting
And the actual spec: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#httpingresspath-v1beta1-extensions
The spec hasn't changed from 1.10 to 1.14 for Ingress.

### Steps To Reproduce

1. Install Kong ingress controller 0.4.0 with Kong 1.1.2
2. Create an example ingress resource with no explicit Path configured (for example:
```
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: name-virtual-host-ingress
spec:
  rules:
  - host: foo.bar.com
    http:
      paths:
      - backend:
          serviceName: service1
          servicePort: 80
```
3. Lots of errors in the Ingress controller and no services registered in Kong

**Comment by jlarocque88** [comment]
I am experiencing the same thing on k8s 1.11.6

**Comment by hbagdi** [comment]
I can confirm this bug. Expect it to be fixed with the upcoming 0.5.0 version.
